codec-http2/src/main/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributor.java

```diff
@@ -211,19 +366,109 @@ int streamableBytes0(Http2Stream stream) {
     }
 
     /**
+     * For testing only!
+     */
+    boolean isChild(int childId, int parentId, short weight) {
+        State parent = state(parentId);
+        State child;
+        return parent.children.containsKey(childId) &&
+                (child = state(childId)).parent == parent && child.weight == weight;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract the assignment out of this expression. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AAssignmentInSubExpressionCheck)